openBCISample.js

```diff
@@ -3,63 +3,70 @@ var gaussian = require('gaussian');
 var outliers = require('outliers');
 var stats = require('scientific-statistics');
 
+var k = require('./openBCIConstants');
+
 /** Constants for interpreting the EEG data */
 // Reference voltage for ADC in ADS1299.
 //   Set by its hardware.
 const ADS1299_VREF = 4.5;
-// Assumed gain setting for ADS1299.
-//   Set by its Arduino code.
-const ADS1299_GAIN = 24.0;
 // Scale factor for aux data
 const SCALE_FACTOR_ACCEL = 0.002 / Math.pow(2,4);
-// Scale factor for channelData
-const SCALE_FACTOR_CHANNEL = ADS1299_VREF / ADS1299_GAIN / (Math.pow(2,23) - 1);
+// X, Y, Z
+const ACCEL_NUMBER_AXIS = 3;
+// Default ADS1299 gains array
 
+// For computing Goertzel Algorithm
+// See: http://www.embedded.com/design/configurable-systems/4024443/The-Goertzel-Algorithm
+// This is N
+const GOERTZEL_BLOCK_SIZE = 100;
+const GOERTZEL_K_250 = Math.floor(0.5 + ((GOERTZEL_BLOCK_SIZE * k.OBCILeadOffFrequencyHz) / k.OBCISampleRate250));
+const GOERTZEL_W_250 = ((2 * Math.PI) / GOERTZEL_BLOCK_SIZE) * GOERTZEL_K_250;
+const GOERTZEL_COEFF_250 = 2 * Math.cos(GOERTZEL_W_250);
 
-var k = require('./openBCIConstants');
 
-module.exports = {
-    convertPacketToSample: function (dataBuf) {
-        var self = this;
-        if(dataBuf === undefined || dataBuf === null) {
-            return;
-        }
-        var numberOfBytes = dataBuf.byteLength;
-        var scaleData = true;
-
-        if (numberOfBytes != k.OBCIPacketSize) return;
-        if (dataBuf[0] != k.OBCIByteStart) return;
-        if (dataBuf[32] != k.OBCIByteStop) return;
-
-        var channelData = function () {
-            var out = [];
-            var count = 0;
-            for (var i = 2; i <= numberOfBytes - 10; i += 3) {
-                var number = self.interpret24bitAsInt32(dataBuf.slice(i, i + 3));
-                out.push(number * SCALE_FACTOR_CHANNEL);
-                count++;
-            }
-            return out;
-        };
+var sampleModule = {
 
-        var auxData = function () {
-            var out = [];
-            var count = 0;
-            for (var i = numberOfBytes - 7; i < numberOfBytes - 1; i += 2) {
-                out.push(scaleData ? self.interpret16bitAsInt32(dataBuf.slice(i, i + 2)) * SCALE_FACTOR_ACCEL : self.interpret16bitAsInt32(dataBuf.slice(i, i + 2)));
-                count++;
+    /**
+     * @description This takes a 33 byte packet and converts it based on the last four bits.
+     *                  0000 - Standard OpenBCI V3 Sample Packet
```

Comment:
So this is sort of forward looking, but a stop byte with 0xC0 indicates a standard packet but a stop byte of 0xC1 could mean handle differently such as a time synced packet or a completely custom packet.